core/ipc/sink/aws-sqs-impl/src/main/java/org/opennms/core/ipc/sink/aws/sqs/AmazonSQSMessageConsumerManager.java

```diff
@@ -0,0 +1,192 @@
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.opennms.core.ipc.sink.api.Message;
+import org.opennms.core.ipc.sink.api.MessageConsumerManager;
+import org.opennms.core.ipc.sink.api.SinkModule;
+import org.opennms.core.ipc.sink.common.AbstractMessageConsumerManager;
+import org.opennms.core.logging.Logging;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.InitializingBean;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.AmazonSQSException;
+import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
+/**
+ * The Class AwsMessageConsumerManager.
+ * 
+ * @author <a href="mailto:agalue@opennms.org">Alejandro Galue</a>
+ */
+public class AmazonSQSMessageConsumerManager extends AbstractMessageConsumerManager implements InitializingBean {
+
+    /** The Constant LOG. */
+    private static final Logger LOG = LoggerFactory.getLogger(AmazonSQSMessageConsumerManager.class);
+
+    /** The consumer runners by module. */
+    private final Map<SinkModule<?, Message>, List<AwsConsumerRunner>> consumerRunnersByModule = new ConcurrentHashMap<>();
+
+    /** The thread factory. */
+    private final ThreadFactory threadFactory = new ThreadFactoryBuilder()
+            .setNameFormat("aws-consumer-%d")
+            .build();
+
+    /** The executor. */
+    private final ExecutorService executor = Executors.newCachedThreadPool(threadFactory);
+
+    /** The AWS configuration. */
+    private final Properties awsConfig = new Properties();
+
+    /**
+     * The Class AwsConsumerRunner.
+     */
+    private class AwsConsumerRunner implements Runnable {
+
+        /** The module. */
+        private final SinkModule<?, Message> module;
+
+        /** The closed. */
+        private final AtomicBoolean closed = new AtomicBoolean(false);
+
+        /** The SQS Object. */
+        private final AmazonSQS sqs;
+
+        /**
+         * Instantiates a new AWS consumer runner.
+         *
+         * @param module the module
+         * @throws AmazonSQSException the Amazon SQS exception
+         */
+        public AwsConsumerRunner(SinkModule<?, Message> module) throws AmazonSQSException {
+            this.module = module;
+            sqs = AmazonSQSUtils.createSQSObject(awsConfig);
+            AmazonSQSUtils.ensureQueueExists(awsConfig, sqs, AmazonSQSUtils.getQueueName(module));
+        }
+
+        /* (non-Javadoc)
+         * @see java.lang.Runnable#run()
+         */
+        @Override
+        public void run() {
+            Logging.putPrefix(MessageConsumerManager.LOG_PREFIX);
+            while (!closed.get()) {
+                final String queueUrl = sqs.getQueueUrl(AmazonSQSUtils.getQueueName(module)).getQueueUrl();
+                sqs.receiveMessage(queueUrl).getMessages().forEach(m -> {
```

Comment:
Can we add a comment about the blocking behavior of `receiveMessage` here? From what I can gather it should block for `ReceiveMessageWaitTimeSeconds` and return an empty list if no messages were available.

Reply:
Will do it. The `MessageWaitTimeSeconds` is configured at queue level with my current design. I'll add that as well on the comment.